src/Symfony/Bundle/FrameworkBundle/Resources/config/http_client.xml

```diff
@@ -16,5 +16,16 @@
             <argument type="service" id="http_client" />
         </service>
         <service id="Psr\Http\Client\ClientInterface" alias="psr18.http_client" />
+
+        <service id="scoping_http_client" class="Symfony\Component\HttpClient\ScopingHttpClient">
+            <argument type="service" id="http_client" />
+            <argument type="collection" />
+        </service>
+        <service id="Symfony\Contracts\HttpClient\HttpClientInterface $scopingHttpClient" alias="scoping_http_client" />
```

Comment:
So, a scoped HTTP client is the result, but `$scopingHttpClient`does not sound good here.

Reply:
I thought that `$scopedHttpClient` sounded like it was already "scoped". But really, this gives you a big object that is aware of *all* of the scopes, and has the *ability* to scope, based on the URI. The naming came from this thinking, but maybe it's still confusing?

Reply:
What Ryan said. The same reasoning is why ScopingHttpClient, the class, is named that way also.